monai/transforms/croppad/dictionary.py

```diff
@@ -420,16 +422,21 @@ class CenterScaleCropd(MapTransform, InvertibleTransform):
     def __init__(
         self, keys: KeysCollection, roi_scale: Union[Sequence[float], float], allow_missing_keys: bool = False
     ) -> None:
-        super().__init__(keys, allow_missing_keys=allow_missing_keys)
+        super(CenterSpatialCropd, self).__init__(keys, allow_missing_keys=allow_missing_keys)
         self.roi_scale = roi_scale
 
     def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.ndarray]:
         d = dict(data)
+        # use the spatial size of first image to scale, expect all images have the same spatial size
         img_size = data[self.keys[0]].shape[1:]
         ndim = len(img_size)
         roi_size = [ceil(r * s) for r, s in zip(ensure_tuple_rep(self.roi_scale, ndim), img_size)]
-        sp_crop = CenterSpatialCropd(self.keys, roi_size=roi_size, allow_missing_keys=self.allow_missing_keys)
-        return sp_crop(data=data)
+        cropper = CenterSpatialCrop(roi_size)
```

Comment:
This modification has the problem of mixing the composition (CenterSpatialCrop) and inheritance (CenterSpatialCropd) patterns. It'll be an issue when we implement https://github.com/Project-MONAI/MONAI/issues/1749. I suggest we avoid the inheritance, implement an inverse method here.

Reply:
OK, thanks for raising this point. Let me submit another quick PR to drop the inheritance and add inverse.
Thanks.